File: src/Core/ServiXpress.Domain/Usuario.cs
using Microsoft.AspNetCore.Identity;

namespace ServiXpress.Domain
{
    public class Usuario : IdentityUser
    {
        public string? Nombre { get; set; }
        public string? Apellidos { get; set; }
        public string? Telefono { get; set; }
        public string? Calle { get; set; }
        public int? NumExterior { get; set; }
        public int? NumInterior { get; set; }
        public string? ColoniaFraccionamiento { get; set; }
        public string? CodigoPostal { get; set; }
        public string? Municipio { get; set; }
        public string? Estado { get; set; }
        public string? Descripcion { get; set; }
      
        public DateTime? FechaHoraRegistro { get; set; }

        public string? Estatus { get; set; }
        public EstatusUsuario? EstatusUsuario { get; set; }


        // UNO A MUCHOS
        public virtual ICollection<Calificacion>? CalificacionesHechas { get; set; }
        public virtual ICollection<Calificacion>? CalificacionesRecibidas { get; set; }


        public ICollection<Reporte>? Reportes { get; set; }




        // PUEDE CONTENER LA IMAGEN DE PERFIL DEL USUARIO
        public string? AvatarUrl { get; set; }


        public List<Documento> Documentos { get; set; }
    }
}

File: src/APPMovil/src/Interfaces/Usuario.ts
import { ImageRequestFormData } from "./DOMInterfaces";

export interface EstatusUsuario {
    estatus: string
}

export interface Usuario {
    id: string;
    nombre: string;
    apellidos: string;
    telefono?: string;
    avatarUrl?: string;
    userName?: string;
    email: string;
    password?: string;
    calle?: string;
    codigoPostal?: string;
    coloniaFraccionamiento?: string;
    descripcion?: string;
    estado?: string;
    estatus: string;
    fechaHoraRegistro: Date;
    municipio?: string;
    numExterior?: number;
    numInterior?: number;
    rol: number;
    roles: string[]
    rolNombre: string;
    foto?: any;
    token: string
}


export interface RegisterUser {
    Nombre: string;
    Apellidos: string;
    Telefono: string;
    Rol: number;
    Email: string;
    Password: string;
    Foto?: ImageRequestFormData
}


export interface UpdateUser {
    nombre: string;
    apellidos: string;
    telefono: string;
    calle: string
    rol: number;
    coloniaFraccionamiento: string
    codigoPostal: string
    numExterior: number
    numInterior: number
    municipio: string
    estado: string
    descripcion: string
    email: string;
    password: string;
    foto?: ImageRequestFormData
}
